libbeat/logp/core.go

```diff
@@ -98,10 +99,14 @@ func Configure(cfg Config) error {
 			selectors["*"] = struct{}{}
 		}
 
-		if _, enabled := selectors["stdlog"]; !enabled {
+		_, stdlogEnabled := selectors["stdlog"]
+		_, allEnabled := selectors["*"]
+		if !stdlogEnabled || !allEnabled {
 			// Disable standard logging by default (this is sometimes used by
 			// libraries and we don't want their spam).
 			golog.SetOutput(ioutil.Discard)
```

Comment:
IIUC the goal is to by default set the output of the `log` package to `ioutil.Discard`. So this call should happen always. In it's current location it only happens if we are running with `logging.level: debug and len(logging.selectors) > 0`. I think it needs be moved outside the `if` block.

Then the case were we do want to allow the `log` pkg to output can happen here.

Reply:
I think its more like this:

```golang

// If we are in debug mode and we have the all selectors actived we also redirect the golog output.
_, allEnabled := selectors["*"]
if cfg.Level.Enabled(DebugLevel) && allEnabled { // "*"
    golog.SetOutput(_defaultGoLog)
} else {
    golog.SetOutput(ioutil.Discard) 
}

```
The other case is when we are in debug mode + stdlog selector activated.
``` golang
/// specific _stdlog_ selector when debug is on.
, stdlogEnabled := selectors["stdlog"]
if !stdlogEnabled {
golog.SetOutput(_defaultGoLog)
}
```

Reply:
Added the CLI flag to the above command to hopefull better understand how a human would use this.